src/r-bridge/shell-executor.ts

```diff
@@ -0,0 +1,132 @@
+import {DEFAULT_R_SHELL_OPTIONS, RShellExecutionOptions} from './shell'
+import {deepMergeObject} from '../util/objects'
+import {spawnSync} from 'child_process'
+import {parseCSV, ts2r} from './lang-4.x'
+import {SemVer} from 'semver'
+import semver from 'semver/preload'
+import type {ILogObj, Logger} from 'tslog'
+import {log} from '../util/log'
+import fs from 'fs'
+import {removeTokenMapQuotationMarks, TokenMap} from './retriever'
+import {DeepWritable} from 'ts-essentials'
+
+export class RShellExecutor {
+
+	public readonly options:        Readonly<RShellExecutionOptions>
+	private readonly log:           Logger<ILogObj>
+	private readonly prerequisites: string[] = []
```

Comment:
Maybe allow to add arbitrary commands as prerequisites? 
In a way it may be nice to toggle some of those in the future? 

Yet, this can be outsourced into another issue :)